app/dao/notification_history_dao.py

```diff
@@ -33,3 +33,18 @@ def delete_notification_history_older_than_datetime(older_than_datetime, query_l
     current_app.logger.info(
         "Deleted %s rows from notification_history older than %s", total_deleted, older_than_datetime
     )
+
+
+def delete_notification_history_between_two_datetimes(start, end):
```

Comment:
```suggestion
def delete_notification_history_between_two_datetimes(start: datetime, end: datetime):
```

Also wouldn't hate some type hints on these. Not super sure if these _are_ datetimes or they end up as strings after the celery serialization...

Reply:
I've added this in and rebased it into my original commit